README.md

```diff
@@ -34,35 +34,62 @@ You can either [clone the repository from GitHub](https://github.com/jtebert/kil
 
 ### Dependencies
 
-[HDF5](https://portal.hdfgroup.org/display/HDF5/HDF5), [SFML](https://www.sfml-dev.org/index.php), and [OpenMP](https://www.openmp.org/). Install on Ubuntu-like systems with: `sudo apt install libsfml-dev libhdf5-dev libomp-dev`
+[HDF5](https://portal.hdfgroup.org/display/HDF5/HDF5), [SFML](https://www.sfml-dev.org/index.php), and [OpenMP](https://www.openmp.org/). Install on Ubuntu-like systems with:
 
-[nlohmann/json](https://github.com/nlohmann/json): Add [`json.hpp`](https://github.com/nlohmann/json/releases/download/v3.5.0/json.hpp) to the `include/` folder. **Note:** This links to the most recent release (3.5) that has been verified for compatibility. You can find other [versions and release notes on Github](https://github.com/nlohmann/json/releases).
+    sudo apt install libsfml-dev libhdf5-dev libomp-dev
 
 ### Compile
 
-Create a static library at `bin/libKilosim.a`:
+To compile, copy and paste the following:
 
-    make
-or
+    mkdir build                                #Create a build directory
+    cd build                                   #Move into build directory
+    cmake -DCMAKE_BUILD_TYPE=RelWithDebInfo .. #Configure for your system
+    make                                       #Build the library
+    make install                               #Install the library to the source directory
+    cd ..                                      #Move back to source directory
 
-    make static
+Alternatively, you can run:
 
-Create an executable at `bin/kilosim` using the example `MyKilobot.cpp` and `test.cpp`:
+    make kilosim                        #Build library
+    make examples                       #Build examples (optional)
+    make kilosim_docs                   #Build documentation (optional)
 
-    make exec
+To clean up the build enter the source directory and run:
 
-**NOTE:** This is tested with g++ 7.3 but currently fails (for as-of-yet unclear reasons) with g++ 8.1
+    rm -rf build bin lib
+
+Flags:
+
+ * `-DSANITIZE_ADDRESS=On`: Use address sanitizer to check memory safety
+
+Use flags like so:
+
+    cmake -DSANITIZE_ADDRESS=ON ..
+
+Compatibility:
+
+ * g++ 5   - doesn't work
+ * g++ 7.3 - works
+ * g++ 8.1 - fails due to compiler issue(?)
+ * g++ 8.3 - works 
 
 ### Run
 
-Run the executable generated by `make exec`
+Run the examples generated by `make examples`
 
-    ./bin/kilosim
+    ./bin/kilosim_example examples/exampleConfig.json
 
 ### Using static library
 
-**TODO:** Write tutorial on linking to static library
+If your project uses `cmake`, then incorporating kilosim is easy!
```

Comment:
Do you have a skeleton/example of what a project using this with cmake would look like? Or how to set up/create a cmake project using this?

I don't actually know how to set one up to test this...

Reply:
Sure, you'd make a file name `CMakeLists.txt` with contents like so:
```
cmake_minimum_required (VERSION 3.9)

project(uses_kilosim LANGUAGES CXX)

add_subdirectory(path/to/kilosim)

add_executable(uses_kilosim my_src/my_code.cpp)
target_link_libraries(uses_kilosim PRIVATE kilosim)

#Compile with:
#mkdir build
#cd build
#cmake ..
#make
```
cmake figures out how to compile and link everything together.

Reply:
I feel like I'm missing something here... This alone doesn't generate a Makefile, so I can't `make`?

Reply:
Correct, you need to go through the standard cmake build process: see the comments at the end of the code block above. `cmake ..` generates a makefile (or Ninja file - a number of build systems are supported).

Reply:
Turns out that was me failing at reading error messages -- my bad. I just created a [kilosim-demo](https://github.com/jtebert/kilosim-demo) repository for trying out using the library externally

When I just had a dummy main function, it compiled. But when I copied in code I was using with the non-cmake kilosim (which actually uses the library), it seems like it's not finding/can't link the library:

```
/home/jtebert/google-drive/Harvard/SSR/kilobots/kilosim-test-cmake/src/BayesBot.cpp:9:10: fatal error: Kilobot.h: No such file or directory
 #include <Kilobot.h>
          ^~~~~~~~~~~
compilation terminated.
CMakeFiles/project_name.dir/build.make:62: recipe for target 'CMakeFiles/project_name.dir/src/BayesBot.cpp.o' failed
make[2]: *** [CMakeFiles/project_name.dir/src/BayesBot.cpp.o] Error 1
CMakeFiles/Makefile2:67: recipe for target 'CMakeFiles/project_name.dir/all' failed
make[1]: *** [CMakeFiles/project_name.dir/all] Error 2
Makefile:129: recipe for target 'all' failed
make: *** [all] Error 2
```

Reply:
Nevermind -- dumb mistake. Fixed that error by changing to `#include <kilosim/Kilobot.h>` and now debugging new errors.

Reply:
`incbeta` isn't in the `kilosim` library, so the issue is with the linking in your demo project. Is the demo on Github?

Reply:
It looks as though you're not including the header files, though. You probably want:
```
target_include_directories(bayesbot PUBLIC api)
```
where `api` is a directory containing: `api/bayesbot/incbeta.h`

Reply:
Yeah, the problem is definitely with `CMakeLists.txt` in the demo project. (Code is in [this repo](https://github.com/jtebert/kilosim-demo).) I kept the existing code structure I had from the old version of the code, but not sure what the convention is for project structure.